File: Leetcode/1060-MissingElementInSortedArray/go/solution1.go
package main

import "fmt"

func missingElement(nums []int, k int) int {
	if len(nums) == 1 {
		return nums[0] + k
	}
	for i := 1; i < len(nums); i++ {
		diff := nums[i] - nums[i-1] - 1
		if k-diff <= 0 {
			return nums[i-1] + k
		}
		k -= diff
	}
	return nums[len(nums)-1] + k
}

func main() {
	nums := []int{4, 7, 9, 10}
	k := 3
	res := missingElement(nums, k)
	fmt.Println(res)
}

File: Leetcode/0259-3SumSmaller/cpp/solution1.cpp
#include <algorithm>
#include <vector>

using std::sort;
using std::vector;

class Solution {
 public:
  int threeSumSmaller(vector<int>& nums, int target) {
    // Edge case
    if (nums.size() < 3) {
      return 0;
    }

    int res = 0;
    std::sort(nums.begin(), nums.end());
    for (int i = 0; i < nums.size() - 2; i++) {
      int j = i + 1, k = nums.size() - 1;
      while (j < k) {
        int sum = nums[i] + nums[j] + nums[k];
        if (sum < target) {
          res += k - j;
          j++;
        } else {
          k--;
        }
      }
    }
    return res;
  }
};
